### PR TITLE
Set up persona support

### LINKED ISSUE #23: Create standard festival profiles and allow to start with a scenario

### PR DESCRIPTION
**Pull request by Clemog**
fix #23 

- [x] set up personas support
- [x] add modal to choose the persona
- [ ] add a `safeGetSituation` behaviour (by adding the function from ngc to `@publicodes/tools`)
- [x] refactor flags pattern matching (with decoder ?)

> [!WARNING]
> Seems to have some drops on performance: sliders aren't as fluid as they are currently in prod. Howerver, I'm not sure of the reason yet.

### CODE REVIEW

File: src/Main.elm [hunk 1]
```diff
@@ -89,50 +92,65 @@ emptyModel =
...
-        ( Err e, _, _ ) ->
-            ( { emptyModel | currentError = Just (DecodeError e) }, Cmd.none )
-
-        ( _, Err e, _ ) ->
+        Err e ->
             ( { emptyModel | currentError = Just (DecodeError e) }, Cmd.none )
 
-        ( _, _, Err e ) ->
-            ( { emptyModel | currentError = Just (DecodeError e) }, Cmd.none )
+        Ok rawRules ->
```
Comment (Clemog) [inline src/Main.elm:123]: Je vois pas comment faire autrement 😓

Comment (EmileRolley) [reply src/Main.elm:123]: Je pense que le plus simple serait d'implémenter un `flagsDecoder` :thinking:

Comment (EmileRolley) [reply src/Main.elm]: Done

File: src/Effect.elm [hunk 2]
```diff
@@ -25,6 +25,12 @@ port updateSituation : ( P.RuleName, Json.Encode.Value ) -> Cmd msg
 port scrollTo : ( Int, Int ) -> Cmd msg
 
 
+port showModal : String -> Cmd msg
+
+
+port closeModal : String -> Cmd msg
```
Comment (Clemog) [inline src/Effect.elm:31]: Pourquoi on doit ajouter des ports ?

Comment (EmileRolley) [reply src/Effect.elm:31]: Pour pouvoir appeler des méthodes sur les éléments du DOM.

**Comment by EmileRolley** [comment]
Closed in favor of #31

**Pull request closed without merging**